waspc/cli/src/Wasp/Cli/Command/Build.hs

```diff
@@ -58,40 +73,65 @@ build = do
-        (waspProjectDir </> srcDirInWaspProjectDir)
-        (buildDir </> castRel srcDirInWaspProjectDir)
+    prepareFilesNecessaryForDockerBuild waspProjectDir buildDir = runExceptT $ do
+      -- Until we implement the solution described in https://github.com/wasp-lang/wasp/issues/1769,
+      -- we're copying all files and folders necessary for the build into the .wasp/build directory.
+      -- We chose this approach for 0.12.0 (instead of building from the project root) because:
+      --   - The build context remains small (~1.5 MB vs ~900 MB).
+      --   - We don't risk copying possible secrets from the project root into the build context.
+      --   - The commands for building the project stay the same as before
+      --     0.12.0, which is good for both us (e.g., for fly deployment) and our
+      --     users  (no changes in CI/CD scripts).
+      -- For more details, read the issue linked above.
+      liftIO $
+        copyDirectory
+          (waspProjectDir </> srcDirInWaspProjectDir)
+          (buildDir </> castRel srcDirInWaspProjectDir)
+
+      liftIO $
+        copyDirectory
+          (waspProjectDir </> dotWaspDirInWaspProjectDir </> generatedCodeDirInDotWaspDir </> sdkRootDirInGeneratedCodeDir)
+          (buildDir </> sdkRootDirInGeneratedCodeDir)
+
+      let packageJsonInBuildDir = buildDir </> castRel packageJsonInWaspProjectDir
+      let packageLockJsonInBuildDir = buildDir </> castRel packageLockJsonInWaspProjectDir
+
+      liftIO $
+        copyFile
+          (waspProjectDir </> packageJsonInWaspProjectDir)
+          packageJsonInBuildDir
+
+      liftIO $
+        copyFile
+          (waspProjectDir </> packageLockJsonInWaspProjectDir)
+          packageLockJsonInBuildDir
 
-      copyDirectory
-        (waspProjectDir </> dotWaspDirInWaspProjectDir </> generatedCodeDirInDotWaspDir </> sdkRootDirInGeneratedCodeDir)
-        (buildDir </> sdkRootDirInGeneratedCodeDir)
+      -- A hacky quick fix for https://github.com/wasp-lang/wasp/issues/2368
+      -- We should remove this code once we implement a proper solution.
+      ExceptT $ updateJsonFile removeWaspConfigDevDependency packageJsonInBuildDir
+      ExceptT $ updateJsonFile removeAllMentionsOfWaspConfig packageLockJsonInBuildDir
 
-      copyFile
-        (waspProjectDir </> packageJsonInWaspProjectDir)
-        (buildDir </> castRel packageJsonInWaspProjectDir)
+    removeAllMentionsOfWaspConfig :: Value -> Value
+    removeAllMentionsOfWaspConfig packageLockJsonObject =
+      packageLockJsonObject
+        & key "packages" . key "" %~ removeWaspConfigDevDependency
+        & key "packages" . _Object
+          %~ HM.filterWithKey
+            (\packageKey _ -> not ("wasp-config" `isInfixOf` packageKey))
 
-      copyFile
-        (waspProjectDir </> packageLockJsonInWaspProjectDir)
-        (buildDir </> castRel packageLockJsonInWaspProjectDir)
+    removeWaspConfigDevDependency :: Value -> Value
+    removeWaspConfigDevDependency original =
```

Comment:
Similar applies as for function above.

Ok interesting, this function is a bit more general then the one above -> the one above operates on packageLockJson, while this one operates on any object that has devDependecies in it. I would, similar like for function above, make it clear what this function expects to operate on, in its name, and if not, at least in the comment.

Reply:
Yeah, I actually reuse this one inside `removeAllMentionsOfWaspConfig`.

Which name would you suggest for this one? `removeWaspConfigFromDevDependenciesArray`?

Reply:
Yup, took me a bit to figure out it is reused. Yes, that sounds like ok name.